src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -282,6 +282,34 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
+  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = workflowLoggerWithCall(backendCall)
+    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
+      Try(gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log))) match {
+        case Success(attempts) if attempts.forall(_.isSuccess) =>
+          postProcess(backendCall) match {
+            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
+            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
+              // TODO: What can we return here to retry this operation?
+              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
+              log.error(error, ex)
+              FailedExecutionHandle(new Throwable(error, ex))
+            case Failure(ex) => FailedExecutionHandle(ex)
+          }
+        case Success(attempts) =>
+          TryUtil.sequence(attempts.toSeq) match {
```

Comment:
I do think `ValidationNEL`'s have their place.  My argument is that we haven't really gotten away from re-inventing the wheel.  We're just changing what the wheel looks like.  Because instead of `TryUtil.sequence`, how we just have another function somewhere that does a more abstract version of this:

```scala
val validationNel = nels.toList.sequence[({type λ[a] = ValidationNel[String, a]})#λ, StorageObject]
validationNel.leftMap("Could not copy files: " + _.list.mkString(", ")).toValidationNel
```

So now we've just basically changed the contents of `TryUtil.sequence` to be that.  I know we'll have to extract it anyway, because we already do *this exact same thing* in `WorkflowQueryKey.scala`.  So if I add the above lines of code right now at this very moment it would be irresponsible to not extract it out or we'll be duplicating code.  Where would this code go?  Seems like just a matter of time before we have a `ScalazUtil` object.

> As an analogy would you suggest we always use double instead of ints? Sure we could wrap methods around double to make sure we get int behavior in our specific cases, and the code would probably be pretty simple as well.

No, I would never every suggest that we "always use doubles instead of ints" and I also don't think it's a fair comparison.  In fact I'm arguing for the nuance of choosing double's when double's are necessary and int's when int's are necessary and NOT choosing a double when a simple int would work just fine.  In this case... I don't see how scalaz is adding any value over the standard library.  We haven't got rid of utility code, the return type is not more intuitive, and this chunk of code now has a dependency on a third-party library.